server/src/main/java/org/opensearch/common/blobstore/transfer/RemoteTransferContainer.java

```diff
@@ -173,7 +169,7 @@ private LocalStreamSupplier<InputStreamContainer> getMultipartStreamSupplier(
     }
 
     private boolean isRemoteDataIntegrityCheckPossible() {
-        return isRemoteDataIntegritySupported && !areInputStreamsDecorated;
+        return isRemoteDataIntegritySupported;
```

Comment:
how will we address cases where we need to do data integrity when  input stream is encrypted and remote data integrity is supported  ?

Reply:
We will add relevant check in encryption PR. There was a comment in which use of this flag wasn't clear so we are delegating encryption check for data integrity in the encryption PR.